src/jquery/data.js

```diff
@@ -38,3 +39,42 @@ jQuery.data = function( elem, name, value ) {
 
 	return oldData.apply( this, arguments );
 };
+
+jQuery.fn.extend( {
+    data: function( key, value ) {
```

Comment:
Please use tabs for indentation, not spaces. This applies to most of the added lines.